src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -82,7 +82,9 @@ private function findAndSortTaggedServices(string|TaggedIteratorArgument $tagNam
 
                 if (null !== $indexAttribute && isset($attribute[$indexAttribute])) {
                     $index = $attribute[$indexAttribute];
-                } elseif (null === $defaultIndex && $defaultPriorityMethod && $class) {
+                    $index = $container->getParameterBag()->resolveValue($index) ?: null;
```

Comment:
why making a special case when the resolved value is empty while we were not making a special case when passing an empty string in an attribute value ?
Btw, this change the behavior for this case.

Reply:
The idea was to provide a fall back to `defaultIndexMethod` if the param was set by the env resolver. When I think about it again now this could be resolved by using the "default"  env var processor (`"env(default::FOO)"`) and just check if `$index` is `NULL` and not just falsy.